### PR TITLE
Move split array mark stats to OMR for metronome

### PR DESCRIPTION
**Pull request by jason-hall**
Signed-off-by: Jason Hall <jasonhal@ca.ibm.com>

### CODE REVIEW

**Comment by jason-hall** [comment]
Related to: https://github.com/eclipse/openj9/pull/8232

**Comment by youngar** [comment]
cc @dmitripivkine @amicic

**Comment by amicic** [comment]
Needs a few more spots to be updated: constructor, clear() method and merge.
Follow an example of _objectsMarked.

File: gc/stats/MarkStats.hpp [hunk 1]
```diff
@@ -110,6 +112,12 @@ class MM_MarkStats : public MM_Base
 		,_bytesScanned(0)
 		,_startTime(0)
 		,_endTime(0)
+#if defined(J9MODRON_TGC_PARALLEL_STATISTICS)
+		,_syncStallCount(0)
+		,_syncStallTime(0)
+		,_splitArraysProcessed(0)
+		,_splitArraysAmount(0);
```
Comment (amicic) [inline gc/stats/MarkStats.hpp]: remove semicolon

**Comment by youngar** [comment]
@genie-omr build xlinux

**Comment by youngar** [comment]
Build error:
```
11:51:38  /home/jenkins/workspace/PullRequest-linux_x86-64/gc/stats/MarkStats.hpp: In constructor ‘MM_MarkStats::MM_MarkStats()’:
11:51:38  /home/jenkins/workspace/PullRequest-linux_x86-64/gc/stats/MarkStats.hpp:60:11: error: ‘MM_MarkStats::_endTime’ will be initialized after [-Werror=reorder]
11:51:38    uint64_t _endTime;  /**< Mark end time */
11:51:38             ^
11:51:38  /home/jenkins/workspace/PullRequest-linux_x86-64/gc/stats/MarkStats.hpp:53:12: error:   ‘uintptr_t MM_MarkStats::_syncStallCount’ [-Werror=reorder]
11:51:38    uintptr_t _syncStallCount; /**< The number of times the thread stalled at a sync point */

11:51:38  /home/jenkins/workspace/PullRequest-linux_x86-64/gc/stats/MarkStats.hpp:106:2: error:   when initialized here [-Werror=reorder]
11:51:38    MM_MarkStats() :
11:51:38    ^
11:51:38  cc1plus: all warnings being treated as errors
```

File: gc/stats/MarkStats.hpp [hunk 2]
```diff
@@ -110,6 +112,12 @@ class MM_MarkStats : public MM_Base
 		,_bytesScanned(0)
 		,_startTime(0)
 		,_endTime(0)
+#if defined(J9MODRON_TGC_PARALLEL_STATISTICS)
+		,_syncStallCount(0)
+		,_syncStallTime(0)
+		,_splitArraysProcessed(0)
+		,_splitArraysAmount(0)
+#endif /* J9MODRON_TGC_PARALLEL_STATISTICS */
```
Comment (youngar) [inline gc/stats/MarkStats.hpp]:
```suggestion
#if defined(J9MODRON_TGC_PARALLEL_STATISTICS)
		,_syncStallCount(0)
		,_syncStallTime(0)
		,_splitArraysProcessed(0)
		,_splitArraysAmount(0)
#endif /* J9MODRON_TGC_PARALLEL_STATISTICS */
		,_startTime(0)
		,_endTime(0)
```

**Comment by youngar** [comment]
@genie-omr build xlinux

**Pull request merged**